robot_controllers_interface/include/robot_controllers_interface/gyro_handle.h

```diff
@@ -0,0 +1,118 @@
+ *
+ *     * Redistributions of source code must retain the above copyright
+ *       notice, this list of conditions and the following disclaimer.
+ *     * Redistributions in binary form must reproduce the above copyright
+ *       notice, this list of conditions and the following disclaimer in the
+ *       documentation and/or other materials provided with the distribution.
+ *     * Neither the name of the Fetch Robotics Inc. nor the names of its
+ *       contributors may be used to endorse or promote products derived from
+ *       this software without specific prior written permission.
+ * 
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL FETCH ROBOTICS INC. BE LIABLE FOR ANY
+ * DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF
+ * THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+
+
+// Author: Vincenzo Barbato
+
+
+
+#ifndef ROBOT_CONTROLLERS_INTERFACE_GYRO_HANDLE_H
+#define ROBOT_CONTROLLERS_INTERFACE_GYRO_HANDLE_H
+
+#include <robot_controllers_interface/handle.h>
+#include <geometry_msgs/Vector3.h>
+
+
+
+namespace robot_controllers
+{
+
+
+
+/**
+ * @brief Base class for a gyro handle used to provide controller 
+ *        direct access to gyro data though controller interface.
+ */
+
+class GyroHandle : public Handle
+{
+public:
+
+  GyroHandle()
+  {
+  }
+
+  /** @brief Ensure proper cleanup with virtual destructor. */
+  virtual ~GyroHandle()
+  {
+  }
+
+  /** @brief Get the angular velocities from the gyro */
+  virtual const geometry_msgs::Vector3& getAngularVelocities()const = 0;
+
+  /** @brief Set the angular velocities of the gyro */
+  virtual void setAngularVelocities(const geometry_msgs::Vector3& angularVelocities, bool isGyroDataCurrentlyValid) = 0;
+
+  /** @brief Set the name of the gyro */
+  virtual void setName(const std::string& name) = 0;
+
+  /** @brief Set the type of the gyro */
+  virtual void setType(const std::string& type) = 0;
+
+  /** @brief Get the type of this gyro */
+  virtual const std::string& getType()const = 0;
+
+  /** @brief Get the name of this gyro */
+  virtual const std::string& getName()const = 0;
+
+  /** @brief Get the name of this gyro */
+  virtual std::string getName() = 0;
+
+  /** @brief Get the name of this joint. */
+  virtual bool isGyroDataCurrentlyValid()const = 0;
+
+
+
+protected:
+  
+  bool isGyroDataCurrentlyValid_;
```

Comment:
ROS C++ standard uses underscores for variables and camelCase for methods and classes.
http://wiki.ros.org/CppStyleGuide#Member_variables

is_gyro_data_currently_valid_;